Core/CoreTests/Use Cases/GetAssignmentsTests.swift

```diff
@@ -199,4 +199,52 @@ class GetAssignmentsTests: CoreTestCase {
         XCTAssertEqual(assignment.submissionTypes, [.on_paper, .external_tool])
         XCTAssertEqual(assignment.position, 0)
     }
+
+    func testSortOrderByDueDate2() {
+        let dateC = Date().addDays(2)
+        let dateD = Date().addDays(3)
+
+        let aa = APIAssignment.make([ "id": "2", "course_id": "1", "due_at": nil])
+        let bb = APIAssignment.make([ "id": "3", "course_id": "1", "due_at": nil])
+        let cc = APIAssignment.make([ "id": "4", "course_id": "1", "due_at": dateC, "name": "aaa"])
+        let dd = APIAssignment.make([ "id": "5", "course_id": "1", "due_at": dateD, "name": "ccc"])
+        let ee = APIAssignment.make([ "id": "6", "course_id": "1", "due_at": nil])
+        let ff = APIAssignment.make([ "id": "7", "course_id": "1", "due_at": dateD, "name": "bbb"])
+
+        let a = Assignment.make(["id": "2"])
+        let b = Assignment.make(["id": "3"])
+        let c = Assignment.make(["id": "4"])
+        let d = Assignment.make(["id": "5"])
+        let e = Assignment.make(["id": "6"])
+        let f = Assignment.make(["id": "7"])
+
+       //   must do this so dueAtOrder property gets updated
+        try? a.update(fromApiModel: aa, in: databaseClient, updateSubmission: false)
+        try? b.update(fromApiModel: bb, in: databaseClient, updateSubmission: false)
+        try? c.update(fromApiModel: cc, in: databaseClient, updateSubmission: false)
+        try? d.update(fromApiModel: dd, in: databaseClient, updateSubmission: false)
+        try? e.update(fromApiModel: ee, in: databaseClient, updateSubmission: false)
+        try? f.update(fromApiModel: ff, in: databaseClient, updateSubmission: false)
+
+        let useCase = GetAssignments(courseID: "1", sort: .dueAt)
+
+        let assignments: [Assignment] = databaseClient.fetch(predicate: nil, sortDescriptors: useCase.scope.order)
+        XCTAssertEqual(assignments.count, 6)
+        let order = assignments.map { "\($0.id)" }.joined(separator: " ")
+        //  we don't care about the order of the nil values so trim them at the end of the array
```

Comment:
Because it's a fixture though, they all have the same name because I didn't add a custom name for each.  I do have two items w/ the same date that I'm confident should exercise the sorting by name.  I can add the nil values in anyway though.

Reply:
Oh I missed the two equal dates so that's enough for me. Just wanted the name order to be tested. 👍